### PR TITLE
test: add more test coverage

### PR DESCRIPTION
**Pull request by gregorio-gerardi**
(AI generated)### TL;DR

Added unit tests for utility functions to improve code coverage and reliability.

### What changed?

Added comprehensive test suites for several utility functions:
- `OTelPerformanceManager`: Tests for time calculation methods
- `bulkAddEventListener`: Tests for adding multiple event listeners
- `bulkRemoveEventListener`: Tests for removing multiple event listeners
- `generateUUID`: Tests for UUID generation functionality
- `throttle`: Tests for function throttling behavior
- `withErrorFallback`: Tests for error handling wrapper

Each test suite verifies the core functionality, edge cases, and expected behavior of these utility functions.

### CODE REVIEW

**Comment by gregorio-gerardi** [comment]
* **#205** <a href="https://app.graphite.dev/github/pr/embrace-io/embrace-web-sdk/205?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/embrace-io/embrace-web-sdk/205?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

**Comment by gregorio-gerardi** [comment]
FYI: I just wanted to try https://www.cursor.com/, this PR is me asking the AI model "can you increase the test coverage of this repo?" and committing the results. Because... why not? 🤷‍♂️ they look good, so it just took me like 2 minutes 🙂

File: src/utils/bulkAddEventListener/bulkAddEventListener.test.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    callback = sinon.spy();
+  });
+
+  it('should add event listeners for all specified events', () => {
+    const events = ['click', 'mouseover', 'keydown'];
+    bulkAddEventListener({ target, events, callback });
+
+    events.forEach(event => {
+      target.dispatchEvent(new Event(event));
+      void expect(callback).to.have.been.called;
```
Comment (jpmunz) [inline src/utils/bulkAddEventListener/bulkAddEventListener.test.ts:25]: I'd make this slightly more specific to assert they were only called once

File: src/utils/PerformanceManager/OTelPerformanceManager.test.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    performanceManager = new OTelPerformanceManager(mockClock);
+  });
+
+  it('should use default clock when none provided', () => {
+    const defaultManager = new OTelPerformanceManager();
+    expect(defaultManager).to.be.instanceOf(OTelPerformanceManager);
+  });
+
+  it('should calculate epochMillisFromOriginOffset correctly', () => {
+    const offset = 500;
```
Comment (jpmunz) [inline src/utils/PerformanceManager/OTelPerformanceManager.test.ts:26]: I'd make this something other than 500 to make sure it isn't using `now` like in the other test

File: src/utils/PerformanceManager/OTelPerformanceManager.test.ts [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+  it('should get current time in milliseconds', () => {
+    const result = performanceManager.getNowMillis();
+    expect(result).to.equal(1500); // timeOrigin (1000) + now() (500)
+  });
+
+  it('should get current time in HR time format', () => {
+    const result = performanceManager.getNowHRTime();
+    // HR time is [seconds, nanoseconds]
+    expect(result[0]).to.equal(1); // 1500ms = 1 second
```
Comment (jpmunz) [inline src/utils/PerformanceManager/OTelPerformanceManager.test.ts:39]: comment incorrect

File: src/utils/PerformanceManager/OTelPerformanceManager.test.ts [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+  beforeEach(() => {
+    mockClock = {
+      timeOrigin: 1000,
+      now: sinon.stub().returns(500),
+    };
+    performanceManager = new OTelPerformanceManager(mockClock);
+  });
+
+  it('should use default clock when none provided', () => {
```
Comment (jpmunz) [inline src/utils/PerformanceManager/OTelPerformanceManager.test.ts:20]: test description doesn't seem to match the body, maybe just remove altogether I'm not sure how we'd test without providing the clock?

**Review by jpmunz — COMMENTED** [review]
wow it did a pretty good job

File: src/utils/PerformanceManager/OTelPerformanceManager.test.ts (hunk 2 above)
Comment (gregorio-gerardi) [reply src/utils/PerformanceManager/OTelPerformanceManager.test.ts:26]: good call

File: src/utils/bulkAddEventListener/bulkAddEventListener.test.ts (hunk 1 above)
Comment (gregorio-gerardi) [reply src/utils/bulkAddEventListener/bulkAddEventListener.test.ts:25]: I can make it

```
    events.forEach((event, index) => {
      target.dispatchEvent(new Event(event));
      expect(callback.callCount).to.equal(index + 1);
    })
   ```

to assert the number of calls increment with each event

File: src/utils/PerformanceManager/OTelPerformanceManager.test.ts [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+      timeOrigin: 1000,
+      now: sinon.stub().returns(500),
+    };
+    performanceManager = new OTelPerformanceManager(mockClock);
+  });
+
+  it('should calculate epochMillisFromOriginOffset correctly', () => {
+    const offset = 300;
+    const result = performanceManager.epochMillisFromOriginOffset(offset);
+    expect(result).to.equal(1300); // timeOrigin (1000) + offset (500)
```
Comment (jpmunz) [inline src/utils/PerformanceManager/OTelPerformanceManager.test.ts:23]:
```suggestion
    expect(result).to.equal(1300); // timeOrigin (1000) + offset (300)
```

**Review by jpmunz — APPROVED** [review]

**Comment by gregorio-gerardi** [comment]
### Merge activity

* **Apr 7, 3:35 PM EDT**: A user started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/embrace-io/embrace-web-sdk/205).
